web-nfc.d.ts

```diff
@@ -0,0 +1,90 @@
+// Type definitions for Web NFC
+// Project: https://github.com/w3c/web-nfc
+// Definitions by: Takefumi Yoshii <https://github.com/takefumi-yoshii>
+// TypeScript Version: 3.9
+
+// This type definitions referenced to WebIDL.
+// https://w3c.github.io/web-nfc/#actual-idl-index
+
+interface Window {
+  NDEFMessage: NDEFMessage
+}
+declare class NDEFMessage {
+  constructor(messageInit: NDEFMessageInit)
+  records: ReadonlyArray<NDEFRecord>
+}
+declare interface NDEFMessageInit {
+  records: NDEFRecordInit[]
+}
+
+declare type NDEFRecordDataSource = string | BufferSource | NDEFMessageInit
+
+interface Window {
+  NDEFRecord: NDEFRecord
+}
+declare class NDEFRecord {
+  constructor(recordInit: NDEFRecordInit)
+  readonly recordType: string
+  readonly mediaType?: string
+  readonly id?: string
+  readonly data?: DataView
+  readonly encoding?: string
+  readonly lang?: string
+  toRecords?: () => NDEFRecord[]
```

Comment:
Shall we use `ReadonlyArray` as well there?

Reply:
It's not FrozenArray though, just a sequence

Reply:
Looking at WebIDL, it looks like a sequence.